sasmodels/generate.py

```diff
@@ -902,6 +899,10 @@ def _add_source(source, code, path, lineno=1):
     source.append('#line %d "%s"' % (lineno, path))
     source.append(code)
 
+def read_text(f):
```

Comment:
Alternate solution that allows py3 to continue using pathlib:
```suggestion
# CRUFT: py2.7 support
try:
    from pathlib.Path import read_text
    from pathlib import Path
except ImportError:
    class Path:
        def __init__(self, f):
            self.f = f
        def read_text(self):
            with open(self.f) as fid:
                return fid.read()
```

Reply:
Path is a class not a module, so you can't import from it.  You could maybe monkeypatch the backported pathlib to include read_text but probably not worthwhile.